File: from-0201-to-0300/0264-ugly-number-ii/solution.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        uglies = [1]
        f2, f3, f5 = 0, 0, 0
        curr =  1
        while curr < n:
            new_ugly = min(uglies[f2]*2, uglies[f3]*3, uglies[f5]*5)
            if new_ugly == uglies[f2]*2:
                f2 += 1
            if new_ugly == uglies[f3]*3:
                f3 += 1
            if new_ugly == uglies[f5]*5:
                f5 += 1

            uglies.append(new_ugly)
            curr += 1

        return uglies[-1]

File: from-0201-to-0300/0264-ugly-number-ii/solution.go
func nthUglyNumber(n int) int {
    uglies := []int{1}
    u2, u3, u5 := 0, 0, 0
    curr := 1

    for curr < n {
        newU := min3(uglies[u2] * 2, uglies[u3] * 3, uglies[u5] * 5)
        if newU == uglies[u2] * 2 {
            u2 += 1
        }
        if newU == uglies[u3] * 3 {
            u3 += 1
        }
        if newU == uglies[u5] * 5 {
            u5 += 1
        }
        uglies = append(uglies, newU)
        curr += 1
    }

    return uglies[n - 1]
}

func min3(a, b, c int) int {
    m3 := a 
    if b < m3 {
        m3 = b
    }
    if c < m3 {
        m3 = c
    }
    return m3
}